ekncontent/ekncontent/eknc-engine.c

```diff
@@ -480,14 +474,23 @@ eknc_engine_get_domain_for_app (EkncEngine *self,
   g_return_val_if_fail (app_id && *app_id, NULL);
   g_return_val_if_fail (error == NULL || *error == NULL, NULL);
 
-  if (g_hash_table_contains (self->domains, app_id))
-    return g_hash_table_lookup (self->domains, app_id);
+  EkncDomain *domain = g_hash_table_lookup (self->domains, app_id);
```

Comment:
Good point.

Reply:
Looking at this a bit better, we seem to have a lot of API that assumes you can get an `EkncEngine` instance (typically through `eknc_engine_get_default()`) and then use the `for_app` variants of the API. Should this API also go away?

Reply:
Still depends on whether we need to keep that API for the search provider.

I _think_ as far as e-k-l is concerned, Engine and Domain could just be merged altogether. The separation was necessary because Domain was the DB-version-specific implementation, because e-k-l needed to handle all DB versions that we ever published. But now the flatpak runtime takes care of that abstraction.